spec.html

```diff
@@ -9948,91 +9948,57 @@ <h1>Static Semantics: StringValue</h1>
+      <h1>Keywords and Reserved Words</h1>
+      <p>A <em>keyword</em> is a token that matches |IdentifierName|, but also has a syntactic use; that is, it appears literally, in a `fixed width` font, in some syntactic production. The keywords of ECMAScript include `if`, `while`, `async`, `await`, and many others.</p>
+      <p>A <em>reserved word</em> is an |IdentifierName| that cannot be used as an identifier. Many keywords are reserved words, but some are not, and some are reserved only in certain contexts. `if` and `while` are reserved words. `await` is reserved only inside async functions and modules. `async` is not reserved; it can be used as a variable name or statement label without restriction.</p>
+      <p>This specification uses a combination of grammatical productions and early error rules to specify which names are valid identifiers and which are reserved words. All tokens in the |ReservedWord| list below, except for `await` and `yield`, are unconditionally reserved. Exceptions for `await` and `yield` are specified in <emu-xref href="#sec-identifiers"></emu-xref>, using parameterized syntactic productions. Lastly, several early error rules restrict the set of valid identifiers. See <emu-xref href="#sec-identifiers-static-semantics-early-errors"></emu-xref>, <emu-xref href="#sec-let-and-const-declarations-static-semantics-early-errors"></emu-xref>, <emu-xref href="#sec-for-in-and-for-of-statements-static-semantics-early-errors"></emu-xref>, and <emu-xref href="#sec-class-definitions-static-semantics-early-errors"></emu-xref>. In summary, there are five categories of identifier names:</p>
+      <ul>
+        <li>
+          <p>Those that are always recognized as identifiers, such as `Math`, `window`, `toString`, and `_`;</p>
+        </li>
+        <li>
+          <p>Those that are never allowed as identifiers, namely the |ReservedWord|s listed below except `await` and `yield`;
+        <li>
+          <p>Those that are contextually allowed as identifiers, namely `await` and `yield`;</p>
+        </li>
+        <li>
+          <p>Those that are contextually disallowed as identifiers, in strict mode code: `let`, `static`, `implements`, `interface`, `package`, `private`, `protected`, and `public`;</p>
+        </li>
+        <li>
+          <p>Those that are always allowed as identifiers, but also appear as keywords in syntactic productions: `as`, `async`, `from`, `get`, `of`, `set`, and `target`.</p>
+        </li>
+      </ul>
+      <p>The term <em>conditional keyword</em>, or <em>contextual keyword</em>, is sometimes used to refer to the keywords that fall in the last three categories, and thus can be used as identifiers in some contexts and as keywords in others.</p>
       <h2>Syntax</h2>
       <emu-grammar type="definition">
-        ReservedWord ::
-          Keyword
-          FutureReservedWord
-          NullLiteral
-          BooleanLiteral
+        ReservedWord :: one of
```

Comment:
Perhaps `KeywordOrReservedWord`?

Reply:
That would suggest that all keywords (and reserved words) appear in that production, which is not the case.